src/peer_manager.rs

```diff
@@ -1528,6 +1544,14 @@ impl PeerManager {
         }
     }
 
+    /// Removes the given entry from the routing_table or peer_map, returns the removal details
+    pub fn purge_out_of_sync_peer(&mut self,
+                                  name: &XorName)
+                                  -> Result<RemovalDetails<XorName>, RoutingTableError> {
+        let _ = self.peer_map.remove_by_name(name);
```

Comment:
this is purging the peers which are out_of_sync between peer_man and RT.
hence the state can be not only routing? (like the case we encountered, the state is in AwaitingIdentify)